crates/extensions/c8y_mapper_ext/src/availability/actor.rs

```diff
@@ -0,0 +1,295 @@
+                        .await?;
+                    self.start_heartbeat_timer_if_interval_is_positive(source)
+                        .await?;
+                }
+                RegistrationResult::Update => {
+                    self.start_heartbeat_timer_if_interval_is_positive(source)
+                        .await?;
+                }
+                RegistrationResult::Error(reason) => warn!(reason),
+            },
+            EntityType::Service => {}
+        }
+
+        Ok(())
+    }
+
+    /// Insert a <"device topic ID" - "service topic ID" and "external ID"> pair into the map.
+    /// If @health is provided in the registration message, use the value as long as it's valid as a service topic ID.
+    /// If @health is not provided, use the "tedge-agent" service topic ID as default.
+    /// @id is the only source to know the device's external ID. Hence, @id must be provided in the registration message.
+    fn update_device_service_pair(
+        &mut self,
+        registration_message: &EntityRegistrationMessage,
+    ) -> RegistrationResult {
+        let source = &registration_message.topic_id;
+
+        let result = match registration_message.other.get("@health") {
+            None => {
+                Ok(registration_message.topic_id.to_default_service_topic_id("tedge-agent").unwrap())
+            }
+            Some(raw_value) => {
+                match raw_value.as_str() {
+                    None => Err(format!("'@health' must hold a string value. Given: {raw_value:?}")),
+                    Some(maybe_service_topic_id) => {
+                        EntityTopicId::from_str(maybe_service_topic_id)
+                            .map(|id| id.into())
+                            .map_err(|_| format!("'@health' must be the default service topic schema 'device/DEVICE_NAME/service/SERVICE_NAME'. Given: {maybe_service_topic_id}"))
+                    }
+                }
+            }
+        };
+
+        match result {
+            Ok(service_topic_id) => {
+                match registration_message.external_id.clone() {
+                    None => RegistrationResult::Error(format!("'@id' field is missing. Cannot start availability monitoring for the device '{source}'")),
+                    Some(external_id) => {
+                        match self.device_ids_map
+                            .insert(source.clone(), DeviceIds { service_topic_id, external_id }) {
+                            None => RegistrationResult::New,
+                            Some(_) => RegistrationResult::Update,
+                        }
+                    }
+                }
+            }
+            Err(err) => RegistrationResult::Error(format!("'@health' contains invalid value in {source}. Details: {err}")),
+        }
+    }
+
+    /// Set a new timer for heartbeat
+    /// Caution: the heartbeat interval from config is defined in MINUTES, not seconds
```

Comment:
Using `SecondsOrHumanTime` would help to remove the risk for such unit mismatches.